### DISCUSSION TITLE
Dealing with applications that don't support cursor-shape-v1

### DISCUSSION THREAD

**Discussion by jchv** [hyprwm/Hyprland#5248] [General]
I'm invested a bit into the cursor-shape-v1 issue because it would fix what I consider to be broken behavior in my compositor of choice (SwayWM.) Unfortunately, it looks like GTK is doing the usual GTK thing so it seems like getting consistent cursors across all applications might be hard.

I made [an attempt](https://github.com/jchv/wlcursorfix) at doing things the horrible way, by overriding functions using `LD_PRELOAD`. But honestly, it was a fair bit more complicated than I'd hoped and frankly it's not clear that this path is even a good _temporary_ solution. (Of course, if you want to use it, be my guest, just don't say I didn't warn you.)

That said, since GTK is not going to be cooperative right now, I wonder if there are any other clever workarounds that might be worth looking into. If not, I'm seriously considering just trying to make my own patch less horrific.

(I'm not a Hyprland user, but I figure this is probably the most relevant community to discuss this, so please bear with me.)

Thoughts?

**Comment by vaxerski** [comment]
You simply can't without patching gtk (or your fix)

What I do is avoid gtk and set xcursor theme to an xursor version of my cursor theme. For apps that I still have to use (e.g. wine) I just cope with a blurry cursor on my laptop.
